src/datatypes/uncertainty.ts

```diff
@@ -55,41 +56,48 @@ export class Uncertainty {
   isPoint() {
     // Note: Can't use normal equality, as that fails for Javascript dates
     // TODO: Fix after we don't need to support Javascript date uncertainties anymore
-    const lte = (a: any, b: any) => {
-      if (typeof a !== typeof b) {
-        return false;
-      }
+    const lte = (a: any, b: any): boolean | null => {
+      if (typeof a !== typeof b || a?.constructor?.name !== b?.constructor?.name) return null;
```

Comment:
1. Ok - will update

2. This was for consistency...

Suppose it returns false.  Then consider the code that follows the line in question:
```
      if (typeof a !== typeof b || a?.constructor?.name !== b?.constructor?.name) return false;

      if (typeof a.sameOrBefore === 'function') {
        return a.sameOrBefore(b);
      }
```

If a is some type such that a.sameOrBefore is a function, then return the result of the function operation.  There are several implementations of sameOrBefore throughout the engine, but all start roughly the same with a type check.  For example, here's the implementation for datetime:

```
  sameOrBefore(other: any, precision?: any): boolean | null {
    if (!(other.isDate || other.isDateTime)) {
      return null;
    }
```

So the functional composition of the two is:

```
      if (typeof a !== typeof b || a?.constructor?.name !== b?.constructor?.name) return false;

      if (!(b.isDate || b.isDateTime)) {
        return null;
      }
```

That didn't seem consistent to me.  So I changed the return to null for consistency.

```
      if (typeof a !== typeof b || a?.constructor?.name !== b?.constructor?.name) return null;

      if (!(b.isDate || b.isDateTime)) {
        return null;
      }
```

Functional compositions aside, the choice of `false` or `null` makes no difference on the end result of the `isPoint()` operation because of the null coalescing operator on the lte and gte functions. 

```
    return (
      this.low != null &&
      this.high != null &&
      (lte(this.low, this.high) ?? false) &&
      (gte(this.low, this.high) ?? false)
    );
```

Reply:
Ah.  I see.  Yeah, I buy that.  I like consistency; I just hadn't noticed the inconsistency here.  Thank you.

Reply:
1 is resolved